### PR TITLE
Added prometheus-operator ServiceMonitor

### PR DESCRIPTION
**Pull request by dsexton**
This adds an option to create prometheus-operator [ServiceMonitor](https://github.com/coreos/prometheus-operator/blob/master/Documentation/design.md#servicemonitor) resource.

### CODE REVIEW

File: install/kubernetes/hubble/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      k8s-app: hubble-metrics
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+  endpoints:
+  - port: metrics
+    interval: 10s
+    honorLabels: true
+    path: /metrics
+{{- end }}
```
Comment (gandro) [inline install/kubernetes/hubble/templates/servicemonitor.yaml]: Missing newline

File: install/kubernetes/hubble/values.yaml [hunk 2]
```diff
@@ -33,6 +33,8 @@ metrics:
   enabled:
    - drop
   address: ":6943"
+  serviceMonitor:
```
Comment (gandro) [inline install/kubernetes/hubble/values.yaml]: Would you mind adding a comment that briefly explains what this does?

File: install/kubernetes/hubble/templates/servicemonitor.yaml (hunk 1 above)
Comment (dsexton) [reply install/kubernetes/hubble/templates/servicemonitor.yaml]: Done

File: install/kubernetes/hubble/values.yaml (hunk 2 above)
Comment (dsexton) [reply install/kubernetes/hubble/values.yaml]: Added

**Pull request merged**